Samples/AppLifecycle/Activation/cs-wpf-packaged/CsWpfActivation/MainWindow.xaml.cs

```diff
@@ -0,0 +1,150 @@
+namespace CsWpfActivation
+{
+    public partial class MainWindow : Window
+    {
+
+        public MainWindow()
+        {
+            InitializeComponent();
+        }
+
+        private void OutputMessage(string message)
+        {
+            Dispatcher.BeginInvoke(new Action(() =>
+            {
+                StatusListView.Items.Add(message);
+            }));
+        }
+
+        private void ActivationInfoWasButton_Click(object sender, RoutedEventArgs e)
+        {
+            GetActivationInfoWas();
+        }
+
+        private void ActivationInfoUwpButton_Click(object sender, RoutedEventArgs e)
+        {
+            GetActivationInfoUwp();
+        }
+
+        private void GetActivationInfoWas()
+        {
+            // The Windows App SDK GetActivatedEventArgs returns an AppActivationArguments
+            // object, which exposes the specific activation type via its Data property.
+            WindowsAppSdk.AppActivationArguments args = 
+                WindowsAppSdk.AppInstance.GetCurrent().GetActivatedEventArgs();
+
+            // The Windows App SDK includes the ExtendedActivationKind enum, which
+            // includes and extends the UWP ActivationKind enum.
+            WindowsAppSdk.ExtendedActivationKind kind = args.Kind;
+            OutputMessage($"ActivationKind: {kind}");
+
+            if (kind == WindowsAppSdk.ExtendedActivationKind.Launch)
+            {
+                if (args.Data is ILaunchActivatedEventArgs launchArgs)
+                {
+                    string argString = launchArgs.Arguments;
+                    string[] argStrings = argString.Split();
+                    foreach (string arg in argStrings)
+                    {
+                        if (!string.IsNullOrWhiteSpace(arg))
+                        {
+                            OutputMessage(arg);
+                        }
+                    }
+                }
+            }
+            else if (kind == WindowsAppSdk.ExtendedActivationKind.File)
+            {
+                if (args.Data is IFileActivatedEventArgs fileArgs)
+                {
+                    IStorageItem file = fileArgs.Files.FirstOrDefault();
+                    OutputMessage(file.Name);
+                }
+            }
+            else if (kind == WindowsAppSdk.ExtendedActivationKind.Protocol)
+            {
+                if (args.Data is IProtocolActivatedEventArgs protocolArgs)
+                {
+                    Uri uri = protocolArgs.Uri;
+                    OutputMessage(uri.AbsoluteUri);
+                }
+            }
+            else if (kind == WindowsAppSdk.ExtendedActivationKind.StartupTask)
+            {
+                if (args.Data is IStartupTaskActivatedEventArgs startupArgs)
+                {
+                    OutputMessage(startupArgs.TaskId);
+                }
+            }
+        }
+
+        // Use the UWP version of AppInstance::GetActivatedEventArgs.
+        private void GetActivationInfoUwp()
```

Comment:
While we know the old UWP API works fine for UWP, the new API works for every type (including UWP).  We should consider only using the new API in all of our samples going forward.

Reply:
Yes, the Windows App SDK version should be preferred going forward (all things being equal), although it's also important to call out both options, partly to clarify where there might be any differences, and partly to make it clear that app developers do have options.